flake.nix

```diff
@@ -0,0 +1,84 @@
+{
+  inputs = {
+    systems.url = "github:nix-systems/default";
+    nixpkgs.url = "github:nixos/nixpkgs/nixpkgs-unstable";
+  };
+
+  outputs =
+    {
+      self,
+      nixpkgs,
+      systems,
+    }:
+    let
+      forEachSystem =
+        f: nixpkgs.lib.genAttrs (import systems) (system: f { pkgs = import nixpkgs { inherit system; }; });
+    in
+    {
+      devShells = forEachSystem (
+        { pkgs }:
+        {
+          default =
+            pkgs.mkShell.override
+              {
+                # override clang 16 runtime
+                stdenv = pkgs.lowPrio pkgs.llvmPackages_16.stdenv;
+              }
+              {
+                propagatedBuildInputs = [
+                  pkgs.SDL_compat
+                  pkgs.SDL_image
+                  pkgs.libwebp
+                ];
+                # when using SDL_compat instead of SDL_classic, SDL_mixer isn't correctly
+                # detected, but there is no harm just specifying it
+                buildInputs = with pkgs; [
+                  git
+                  pkg-config
+                  apple-sdk_15
+
+                  # Need to use brew cmake since nix use old one 3.30.5
+                  # which has issues with copy libraries to macOs app
+                  #cmake
```

Comment:
There is issue with cmake for the new macOs due to libSystem shared library which is not exists under filesystem anymore due to cmake while generate macOs appication(directory) cannot copy this library to it.

Brew cmake already updated and don't has this issue.

Nix already update the cmake, but it's not avalaible for now.
We can wait and merge when it will be updated in at least unstable nixpkgs.

https://github.com/NixOS/nixpkgs/pull/363395
Merged yesterday